rust/hyperlane-base/src/contract_sync/eta_calculator.rs

```diff
@@ -24,6 +24,7 @@ pub(crate) struct SyncerEtaCalculator {
 
 impl SyncerEtaCalculator {
     /// Calculate the expected time to catch up to the tip of the blockchain.
+    #[instrument(err)]
```

Comment:
Will this do anything? I thought this just does something when the function returns a Result, but this just returns a Duration

Also because the panic occurs on line 62 or 64 we wouldn't get a log

How about we change this to return a Result and move `Duration::from_secs_f64` to `Duration::try_from_secs_f64` https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.try_from_secs_f64 ?